contract/kernel/kernel.go

```diff
@@ -637,6 +692,42 @@ func (k *Kernel) rollbackUpdateReservedContract(desc *contract.TxDesc) error {
 	return err
 }
 
+func (k *Kernel) runUpdateBlockChainData(desc *contract.TxDesc) error {
+	if k.context == nil || k.context.LedgerObj == nil {
+		return fmt.Errorf("failed to update blockchain data, because no ledger object in context")
+	}
+
+	err := k.validateUpdateBlockChainData(desc)
+	if err != nil {
+		k.log.Warn("runUpdateBlockChainData validate params error")
+		return err
+	}
+
+	txid, _ := desc.Args["txid"].(string)
+	publicKey, _ := desc.Args["publicKey"].(string)
+	sign, _ := desc.Args["sign"].(string)
+	k.log.Info("runUpdateBlockChainData", "txid", txid)
+	err = k.context.LedgerObj.UpdateBlockChainData(txid, hex.EncodeToString(desc.Tx.Txid), publicKey, sign)
+	return err
+}
+
+func (k *Kernel) rollbackUpdateBlockChainData(desc *contract.TxDesc) error {
```

Comment:
Since the function of rollbackUpdateBlockChainData does nothing, why not giving an empty implementation?